catalog/nato/stanag4559/catalog-stanag4559-source/src/test/java/alliance/catalog/nato/stanag4559/source/TestStanag4559FilterDelegate.java

```diff
@@ -0,0 +1,827 @@
+        assertThat(filter, is(getPrimary(Stanag4559FilterFactory.BTW,
+                INT + Stanag4559FilterFactory.COMMA + INT)));
+    }
+
+    @Test
+    public void testPropertyBetweenLongLiterals() {
+        String filter = filterDelegate.propertyIsBetween(PROPERTY, (long) INT, (long) INT);
+        assertThat(filter, is(getPrimary(Stanag4559FilterFactory.BTW,
+                INT + Stanag4559FilterFactory.COMMA + INT)));
+    }
+
+    @Test
+    public void testPropertyBetweenFloatLiterals() {
+        String filter = filterDelegate.propertyIsBetween(PROPERTY, FLOAT, FLOAT);
+        assertThat(filter, is(getPrimary(Stanag4559FilterFactory.BTW,
+                FLOAT + Stanag4559FilterFactory.COMMA + FLOAT)));
+    }
+
+    @Test
+    public void testPropertyBetweenDoubleLiterals() {
+        String filter = filterDelegate.propertyIsBetween(PROPERTY, (double) FLOAT, (double) FLOAT);
+        assertThat(filter, is(getPrimary(Stanag4559FilterFactory.BTW,
+                FLOAT + Stanag4559FilterFactory.COMMA + FLOAT)));
+    }
+
+    @Test
+    public void testPropertyNull() {
+        String filter = filterDelegate.propertyIsNull(PROPERTY);
+        assertThat(filter, is(
+                Stanag4559FilterFactory.NOT + getPrimary(Stanag4559FilterDelegate.EMPTY_STRING,
+                        Stanag4559FilterFactory.EXISTS)));
+    }
+
+    @Test
+    public void testPropertyLike() {
+        String filter = filterDelegate.propertyIsLike(ANY_TEXT, ATTRIBUTE, false);
+
+        List<AttributeInformation> attributeInformationList = generateAttributeInformation().get(
+                Stanag4559Constants.NSIL_ALL_VIEW);
+        StringBuilder result = new StringBuilder(Stanag4559FilterFactory.LP);
+        for (AttributeInformation attributeInformation : attributeInformationList) {
+            if (attributeInformation.attribute_type.equals(AttributeType.TEXT)) {
+                result.append(getPrimary(attributeInformation.attribute_name,
+                        Stanag4559FilterFactory.LIKE,
+                        Stanag4559FilterDelegate.SQ + ATTRIBUTE + Stanag4559FilterDelegate.SQ)
+                        + Stanag4559FilterFactory.OR);
+            }
+
+        }
+        String resultString = result.toString();
+        resultString = resultString.substring(0, result.length() - 4) + Stanag4559FilterFactory.RP;
+        assertThat(filter, is(resultString));
+    }
+
+    @Test
+    public void testPropertyLikeNullQueryableAttributes() {
+        Stanag4559FilterDelegate filterDelegate = new Stanag4559FilterDelegate(null, Stanag4559Constants.NSIL_ALL_VIEW);
+        String filter = filterDelegate.propertyIsLike(ANY_TEXT, ATTRIBUTE, false);
+        assertThat(filter, is(Stanag4559FilterDelegate.EMPTY_STRING));
+    }
+
+    @Test
+    public void testPropertyLikeNullQueryableAttributesForView() {
+        HashMap<String, List<AttributeInformation>> map = new HashMap<>();
+        map.put(Stanag4559Constants.NSIL_ALL_VIEW, null);
```

Comment:
usually I declare as Map<x,y> = new HashMap<>() and pass interface vs implementation.